frontend/src/types.ts

```diff
@@ -689,6 +689,8 @@ export type ImageStreamAndVersion = {
   imageVersion?: ImageStreamSpecTagType;
 };
 
+export type CustomWatchK8sResult<T> = [data: T, loaded: boolean, error?: Error | undefined];
```

Comment:
You should reference the type from `WatchK8sResult`. Note that `K8sResourceCommon | K8sResourceCommon` must come from the SDK and not our own definitions.
A comment as to why this exists would be useful until the SDK gets updated at which point we can remove this.
```suggestion
export type CustomWatchK8sResult<R extends K8sResourceCommon | K8sResourceCommon[]> = [
  data: WatchK8sResult<R>[0],
  loaded: WatchK8sResult<R>[1],
  loadError: Error | undefined,
];
```

Reply:
We already have a definition for `K8sResourceCommon` in `types.ts`. So you want me to move this type declaration to `k8sTypes.ts` file?

Reply:
Our definition is `deprecated`. Since this new type is to work with utils from the SDK, it should use the types from the SDK as well.